### ISSUE TITLE
UWP Video player Crash

### ISSUE THREAD

**Issue by RuddyOne** [inthehand/InTheHand.Forms#52]
`has exited with code -1073741819 (0xc0000005) 'Access violation'.`

I get this error randomly when using the video player for playing local MP4 videos. It can occur at any time using the video player e.g. first load or it could be 50th (of the view appearing). Any idea what is causing this?

Note: I have images in the same location as the videos and they all work 100%, just in case we are thinking the application cannot access its location.

**Comment by peterfoot** [comment]
I don't know why it would cause this. Is there anything in common about the videos which fail - are they particularly large files, same codec etc?

**Comment by RuddyOne** [comment]
So they are all MP4 and the size varies from 80MB to 250MB, I am going to find the codec now. 

**Update**: Codec: H264 - MPEG-4 AVC (part 10) (avc1), same for all videos.

Also this time when it crashed (on a different video it gave me the stack trace). See below.

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at InTheHand.Forms.Platform.WinRT.MediaElementRenderer.Control_CurrentStateChanged(Object sender, RoutedEventArgs e)
   at Windows.ApplicationModel.Core.UnhandledError.Propagate()
   at Microsoft.AppCenter.Utils.ApplicationLifecycleHelper.<.ctor>b__17_1(Object sender, UnhandledErrorDetectedEventArgs eventArgs)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.AppCenter.Utils.ApplicationLifecycleHelper.<.ctor>b__17_1(Object sender, UnhandledErrorDetectedEventArgs eventArgs)
```

Not sure if that is even related as I did not get 'Access violation' in the output?

**Comment by Johno-ACSLive** [comment]
I am getting this issue as well, basically I have a URL to an MP4 video where the URL is no longer valid. If I use a native UWP mediaelement I get a message saying unable to play the file. Using this library in Xamarin.Forms running as a UWP app, my app crashes in this scenario.

If I run the same Xamarin.Forms app on Android I get a message saying the file cannot be played - the expected behaviour.

Can't test my Xamarin.Forms app in iOS at the moment.

EDIT:

OK so I think I found what is triggering the bug. I was setting KeepScreenOn = true, if I comment out this line (I believe the default value is false), the crash no longer occurs.

Process of what's happening for the UWP crash, we set KeepScreenOn = true via code, we set the source to a URL with an MP4 file that is invalid (not available). Media Element tries to load the file, errors out as it should but then we get an app crash. Disabling KeepScreenOn, media element tries to load the file, errors out as it should but in this case no app crash.

I haven't looked at the code but seems to be some logic in the UWP logic that's not handling the state changes correctly when the KeepScreenOn value is set to true.

Hope this helps track down the bug.
